src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/RequestRejectionUtilities.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Http
+{
+    public static class RequestRejectionUtilities
+    {
+        private const int _maxExceptionDetailSize = 128;
+
+        public static void RejectRequest(RequestRejectionReason reason)
+        {
+            throw BadHttpRequestException.GetException(reason);
+        }
+
+        public static void RejectRequest(RequestRejectionReason reason, string detail)
+        {
+            throw BadHttpRequestException.GetException(reason, detail);
+        }
+
+        // Do not move GetException() code into this method.
+        // From https://github.com/aspnet/KestrelHttpServer/pull/1469#issuecomment-285161010:
+        // Best exception pattern is call void method that only throws; and does no other work
+        // (other than calling another function); unless it's a fairly simple 'new Exception'.
+        // The Jit does 3 things with this when called:
```

Comment:
We have a way to verify this programmatically? If not, have we verified this manually?